src/evidently/collector/app.py

```diff
@@ -190,6 +193,10 @@ def is_authenticated(connection: ASGIConnection, _: BaseRouteHandler) -> None:
         if not connection.scope["auth"]["authenticated"]:
             raise NotAuthorizedException()
 
+    async def on_startup(app: Litestar) -> None:
+        # TODO: check task health
+        await loop(seconds=service.check_interval, func=check_snapshots_factory(service, service.storage))
+
```

Comment:
The canonical way of handling long-running tasks in Litestar is to make use of the `lifespan` context managers, which are purpose built for this:

https://docs.litestar.dev/latest/usage/applications.html#lifespan-context-managers

You could define one like so:
```suggestion
@contextlib.asynccontextmanager
def lifespan(app: Litestar) -> AsyncGenerator[None, None]:
    func = check_snapshots_factory(service, service.storage)
    event = asyncio.Event()
    
    async def _loop(app: Litestar):
        while not event.is_set():
            await func()
            await asyncio.sleep(service.check_interval)

    task = asyncio.create_task(_loop())
    try:
        yield 
    finally:
          event.set()
          await task
```

and add it to `Litestar(..., lifespan=[lifespan])`.

Litestar guarantees that things you call there won't be cancelled and will gracefully shutdown, even in case of a server error. The `on_startup` hook, while it can be used to achieve similar things, doesn't explicitly guarantee this and doesn't offer a way to handle graceful shutdowns of the tasks because it's a one-off function, which means you'll end up cancelling your tasks, which can lead to some warnings or potential errors because of unfinished coroutines. It's not a big deal in this case but I thought I'd mention it :)

Alternatively, if you don't want to deal with managing the task manually at all:

```python
@contextlib.asynccontextmanager
def lifespan(app: Litestar) -> AsyncGenerator[None, None]:
    func = check_snapshots_factory(service, service.storage)

    async def _loop():
        while True:
            await func()
            await asyncio.sleep(service.check_interval)

    async with anyio.create_task_group() as tg:
      try:
          tg.start_soon(loop)
          yield 
      finally:
          tg.cancel_scope.cancel()
```